dom0/sd-export.sls

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+# vim: set syntax=yaml ts=2 sw=2 sts=2 et :
+
+#
+# Installs 'sd-export' AppVM, to persistently store SD data
+# This VM has no network configured.
+##
+include:
+  - sd-workstation-template
+
+sd-export-template:
+  qvm.vm:
+    - name: sd-export-template
+    - clone:
+      - source: securedrop-workstation
+      - label: red
+    - tags:
+      - add:
+        - sd-workstation
+    - require:
+      - sls: sd-workstation-template
+
+sd-export-usb-dvm:
+  qvm.vm:
+    - name: sd-export-usb-dvm
+    - present:
+      - template: sd-export-template
+      - label: red
+    - prefs:
+      - netvm: ""
+      - template_for_dispvms: True
+    - tags:
+      - add:
+        - sd-workstation
+    - require:
+      - qvm: sd-export-template
+
+# Ensure the Qubes menu is populated with relevant app entries,
+# so that Nautilus/Files can be started via GUI interactions.
+sd-export-template-sync-appmenus:
+  cmd.run:
+    - name: >
+        qvm-start --skip-if-running sd-export-template &&
+        qvm-sync-appmenus sd-export-template
+    - require:
+      - qvm: sd-export-template
+    - onchanges:
+      - qvm: sd-export-template
+
+# Here we must create as the salt stack does not appear to allow us to create
+# VMs with the class DispVM and attach the usb device specified in the config
+# permanently to this VM
+sd-export-create-named-dispvm:
+  cmd.run:
+    - name: >
+        qvm-check sd-export-usb ||
+        qvm-create --class DispVM --template sd-export-usb-dvm --label red sd-export-usb
+    - require:
+      - qvm: sd-export-usb-dvm
+
+{% import_json "sd/config.json" as d %}
+
+sd-export-named-dispvm-permanently-attach-usb:
+  cmd.run:
+    - name: >
+        qvm-usb attach --persistent sd-export-usb {{ d.usb.device }} || true
```

Comment:
Previous iterations of this step would be run after `qvm-usb detach` on the `sd-export-usb` domain this is because when running `make sd-export` using more than once, with different values for `usb.device` will result in an error while starting the `sd-export-usb` vm (see screenshot below)
![sd-export-error](https://user-images.githubusercontent.com/15223328/58970585-b8aa1600-8787-11e9-85c8-ae48032e1324.png)